core/templates/pages/exploration-editor-page/services/router.service.spec.ts

```diff
@@ -139,7 +139,7 @@ describe('Router Service', () => {
 
   it('should navigate to main tab if path is ""', fakeAsync(() => {
     window.location.hash = '';
-    routerService._changeTab('');
+    routerService._changeTab(undefined);
```

Comment:
@heyimShivam can you give us some insight into this change?

Reply:
yes router.service flake appeared now

Reply:
i have tested this part of code too now 
_changeTab(newPath: string): void {
    if (newPath === undefined) {
      this._changeTab('/');
      return;
    }

Reply:
@gp201 PTAL

Reply:
20 more re-runs to make sure everything is working fine https://github.com/heyimShivam/oppia/runs/7374274416?check_suite_focus=true

Reply:
From what I gather, the original flake was that the code below was not being executed https://github.com/oppia/oppia/blob/40cefaf892508ad4f69411e0369456b16a6caa92/core/templates/pages/exploration-editor-page/services/router.service.ts#L76-L80

and this was because we were passing `''` instead of `undefined`. Please correct me if I'm wrong.

Reply:
yes

Reply:
i think earlier other files seems to test this case

Reply:
@heyimShivam can you confirm if `newPath` is actually 'undefined' during runtime?

Reply:
@kevintab95 can we expect the value of `newPath` to be `''` and `undefined`?
If so can we change the code to:
```
 _changeTab(newPath: string): void { 
   if (newPath === undefined || newPath === '') { 
     this._changeTab('/'); 
     return; 
   } 

```

Reply:
when the user visits the creator-dashboard page. firstly the URL is this https://www.oppia.org/create/4NK8JgNBRChO#
this time the this._changeTab() is called with undefined value

Reply:
![Screenshot from 2022-07-17 07-22-09](https://user-images.githubusercontent.com/77006897/179380676-ed2e6ab9-8f61-4a60-935a-f6c29250fc8f.png)

![Screenshot from 2022-07-17 07-22-00](https://user-images.githubusercontent.com/77006897/179380678-4bcbd39f-be20-4c4f-ba1c-eb6cebc8009d.png)

Reply:
The value can be `undefined` but also add a check for `''` too just like praneeth suggested.

Reply:
As suggested above, there is no harm in adding a check. Maybe we can just do `if (!newPath) {`.